rigid_transformations.py

```diff
@@ -0,0 +1,611 @@
+
+        group_exp_transfo = np.zeros(6)
+        group_exp_transfo[0:3] = rot_vec
+
+        skew_mat = rotations.skew_matrix_from_vector(rot_vec)
+
+        if angle == 0:
+            coef_1 = 0
+            coef_2 = 0
+        elif angle < epsilon:
+            coef_1 = 1. / 6. - angle ** 3 / 120.
+            coef_2 = 1. / 2. - angle ** 2 / 24.
+        else:
+            coef_1 = angle ** (-2) * (1. - np.sin(angle) / angle)
+            coef_2 = angle ** (-2) * (1. - np.cos(angle))
+
+        sq_skew_mat = np.dot(skew_mat, skew_mat)
+        group_exp_transfo[3:6] = (translation
+                                  + coef_1 * np.dot(sq_skew_mat, translation)
+                                  + coef_2 * np.dot(skew_mat, translation))
+
+    else:
+        ref_point = regularize_transformation(ref_point)
+
+        jacobian = jacobian_translation(ref_point, left_or_right='left')
+        inv_jacobian = np.linalg.inv(jacobian)
+
+        tangent_vector_at_identity = np.dot(inv_jacobian, tangent_vector)
+        group_exp_from_identity = group_exp(tangent_vector_at_identity)
+
+        group_exp_transfo = compose(ref_point, group_exp_from_identity)
+
+    return regularize_transformation(group_exp_transfo)
+
+
+def group_log(transfo,
+              ref_point=GROUP_IDENTITY,
+              epsilon=1e-5):
+    """
+    Compute the group logarithm of point transfo,
+    from point ref_point.
+
+    Formula:
+    TODO(nina).
+
+    :param transfo: 6d tangent_vectorector element of SE(3)
+    :param ref_point: 6d tangent_vectorector element of SE(3)
+
+    :returns tangent vector: 6d tangent vector at ref_point.
+    """
+
+    transfo = regularize_transformation(transfo)
+    if ref_point is GROUP_IDENTITY:
+        rot_vec = transfo[0:3]
+        translation = transfo[3:6]
+        angle = np.linalg.norm(rot_vec)
+
+        tangent_vector = np.zeros(6)
+        tangent_vector[0:3] = rot_vec
+
+        if angle == 0:
+            coef_1 = 0
+            coef_2 = 0
+        elif angle < epsilon:
+            coef_1 = .5
+            coef_2 = 1. / 2. - angle ** 2 / 90.
+        else:
+            coef_1 = .5
+            coef_2 = 1 - .5 * angle * np.sin(angle) / (1. - np.cos(angle))
+            coef_2 = coef_2 / angle ** 2
+
+        skew_rot_vec = rotations.skew_matrix_from_vector(rot_vec)
+        sq_skew_rot_vec = np.dot(skew_rot_vec, skew_rot_vec)
+        tangent_vector[3:6] = (translation
+                               - coef_1 * np.dot(skew_rot_vec, translation)
+                               + coef_2 * np.dot(sq_skew_rot_vec, translation))
+
+    else:
+        ref_point = regularize_transformation(ref_point)
+        jacobian = jacobian_translation(ref_point, left_or_right='left')
+        transfo_near_id = compose(inverse(ref_point), transfo)
+        tangent_vector_from_id = group_log(transfo_near_id)
+        tangent_vector = np.dot(jacobian, tangent_vector_from_id)
+
+    return tangent_vector
+
+
+# --- Riemannian
```

Comment:
comment is kinda useless. if you really want to make it clear that these are specific, maybe put this in a riemannian.py file? more of a nit though

Reply:
Yes, actually when I told you that I wanted to factorize code this is because the code on riemannian is general across rotations and rigid transformations. So I will indeed create a separate file riemannian.py and I want to use OO code for lie groups to use the riemannian functions.

I'll only delete the comments for now.

Reply:
you can add a todo if you want